### ISSUE TITLE
createWriteStream and archiver, got `TypeError: undefined is not an object       at node:stream:2060:82` error

### ISSUE THREAD

**Issue by pedroth** [oven-sh/bun#6992]
### What version of Bun is running?

1.0.11

### What platform is your computer?

Ubuntu 22.04.3 LTS

### What steps can reproduce the bug?

Ran the following file:
```js
// index.js

import {createWriteStream} from "fs";
import archiver from "archiver";

const dest = "node_modules.zip";
const src = "./node_modules";

const output = createWriteStream(dest);
const archive = archiver("zip", { zlib: { level: 9 } });

archive.pipe(output);
archive.directory(src, false);
archive.finalize();
```

With the following `package.json`:
```json
{
  "name": "bunbug",
  "module": "index.js",
  "type": "module",
  "devDependencies": {
    "archiver": "^6.0.1",
    "bun-types": "latest"
  },
  "peerDependencies": {
    "typescript": "^5.0.0"
  }
}
```

By running `bun index.js`, I get the following issue:

```
2055 |         state.dataEmitted = !0, this.emit("data", ret);
2056 |       return ret;
2057 |     }, Readable.prototype._read = function(n) {
2058 |       throw new ERR_METHOD_NOT_IMPLEMENTED("_read()");
2059 | 
2060 |     }, Readable.prototype.pipe = function(dest, pipeOpts) {
                                                                                     ^
TypeError: undefined is not an object
      at node:stream:2060:82
      at /home/pedroth/Desktop/bunbug/node_modules/lazystream/lib/lazystream.js:33:4
      at /home/pedroth/Desktop/bunbug/node_modules/lazystream/lib/lazystream.js:18:4
      at /home/pedroth/Desktop/bunbug/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:443:4
      at resume_ (/home/pedroth/Desktop/bunbug/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:790:4)
      at processTicksAndRejections (:61:76)
```

If I run using `node` it works.

### What is the expected behavior?

It should create `node_modules.zip` file.

### What do you see instead?

The following error:

```
2055 |         state.dataEmitted = !0, this.emit("data", ret);
2056 |       return ret;
2057 |     }, Readable.prototype._read = function(n) {
2058 |       throw new ERR_METHOD_NOT_IMPLEMENTED("_read()");
2059 | 
2060 |     }, Readable.prototype.pipe = function(dest, pipeOpts) {
                                                                                     ^
TypeError: undefined is not an object
      at node:stream:2060:82
      at /home/pedroth/Desktop/bunbug/node_modules/lazystream/lib/lazystream.js:33:4
      at /home/pedroth/Desktop/bunbug/node_modules/lazystream/lib/lazystream.js:18:4
      at /home/pedroth/Desktop/bunbug/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:443:4
      at resume_ (/home/pedroth/Desktop/bunbug/node_modules/lazystream/node_modules/readable-stream/lib/_stream_readable.js:790:4)
      at processTicksAndRejections (:61:76)
```

**Comment by Jarred-Sumner** [comment]
Spent some time digging into this.

It's something to do with how the `readable-stream` npm package interacts with our `node:stream` package. It might be caused by `ReadableState` making `pipesCount` a read-only getter. But I'm uncertain.

**Comment by pesa2000** [comment]
I have the exact same issue running Bun v. 1.0.11 and archiver running the same code of @pedroth

**Comment by ChiefORZ** [comment]
same error with the npm package `form-data` which uses `combined-stream` and `delayed-stream`.

**Comment by kartojal** [comment]
anyone got a fix? still happening with `createWriteStream` and `archiver`.

@Jarred-Sumner Sample TS file to debug:

```
import fs from 'fs';
const archiver = require('archiver');

export const zipDir = (dst: string, out: string) => new Promise((resolve, reject) => {

    // create a file to stream archive data to.
    const output = fs.createWriteStream(out);
    const archive = archiver('zip', { zlib: { level: 9 } });

    output.on('close', () => {
        resolve(true);
    });

    archive.on('error', (err) => {
        reject(err);
    });

    // As we're globbing for files in the same dir we've put the archive, we must filter it
    archive.directory(dst, false, file => !file.name.endsWith('.zip') ? file : false);

    archive.pipe(output);

    archive.finalize();
});

// note: before running, create a directory and dummy files: 
// bash: mkdir demo-dir && touch demo-dir/a && touch demo-dir/b

const main = async () => {
   const inputDirectory = "demo-dir";
   const outputZip = "demo.zip";
   
   await zipDir(inputDirectory, outputZip);
}

main();
```

Error with Bun 1.0.19:
```
21156 |     PassThrough.call(this, options);
21157 |     beforeFirstCall(this, "_read", function() {
21158 |       var source = fn.call(this, options);
21159 |       var emit2 = this.emit.bind(this, "error");
21160 |       source.on("error", emit2);
21161 |       source.pipe(this);
              ^
TypeError: undefined is not an object
      at node:stream:2060:83
      at resume_ ()
      at processTicksAndRejections (:61:77)
```

**Comment by D-Sketon** [comment]
Same with `createReadStream`
![image](https://github.com/oven-sh/bun/assets/49871906/028a7d12-833f-4d91-a0dd-c6b9e93d2065)
![image](https://github.com/oven-sh/bun/assets/49871906/9fb57a17-7878-4c20-9952-9409213f6c81)

**Comment by nkarnauh** [comment]
Same error with `node:stream`

<img width="415" alt="image" src="https://github.com/oven-sh/bun/assets/83829392/0292b797-f9a4-40af-a151-e6706e5ad8c2">

**Comment by wolszczak96** [comment]
still an issue in bun 1.1.0

**Comment by wolszczak96** [comment]
still an issue in v1.1.0. Reproduction repo: https://github.com/wolszczak96/bun-streams

**Comment by alec-watts** [comment]
Any fix for this?

**Comment by seveibar** [comment]
I think this was fixed in or before v1.1.7, my test script on https://github.com/oven-sh/bun/issues/10416 successfully runs in v1.1.7

**Comment by pedroth** [comment]
Tested on my use case, and it worked, thank you bun team
